vkubelet/pod.go

```diff
@@ -192,6 +192,10 @@ func (s *Server) updatePodStatus(ctx context.Context, pod *corev1.Pod) error {
 			pod.Status.Reason = "NotFound"
 			pod.Status.Message = "The pod status was not found and may have been deleted from the provider"
 			for i, c := range pod.Status.ContainerStatuses {
+				// Skip waiting and Terminated status
+				if c.State.Waiting != nil || c.State.Terminated != nil {
```

Comment:
Instead of skipping, let's just change how we are setting these values.
When in Terminated or Running we should have a StartedAt, and in Terminated it should also have a proper terminated time as well.

Reply:
Agree. I rewrite the bugfix code (`still skip the waiting state`).